### ISSUE TITLE
enhancement, api: consider create a unify fault type for database filter? 

### ISSUE THREAD

**Issue by qinggniq** [envoyproxy/envoy#15597]
*Title*: *consider create a unify fault type for database filter*

*Description*:
> I'm trying to implement fault injection for mysql proxy, the fault message definition is totally migrated from `RedisFault` https://github.com/envoyproxy/envoy/blob/a240824c376693b16ca8be51e435b95c42f3449f/api/envoy/extensions/filters/network/redis_proxy/v3/redis_proxy.proto#L186-L207
> In the mean while, I notice that in `filters/common/fault/v3` https://github.com/envoyproxy/envoy/blob/a240824c376693b16ca8be51e435b95c42f3449f/api/envoy/extensions/filters/common/fault/v3/fault.proto#L53-L57, 
> The common fault delay is very limit that it cannot let users inject fault base on command type, `mongo filter` uses this fault type, but redis uses its own fault type, I guess they will reimplement their own fault type once some contributors want to extend other filters. It's there any plans to give a unified fault api for those filters?

[optional *Relevant Links*:]
>Any extra documentation required to understand the issue.

**Comment by snowp** [comment]
I don't believe anyone is working on this, but it seems like a reasonable idea at a high level. Marking help wanted in case anyone wants to pick this up
